src/ui_basic/editbox.cc

```diff
@@ -212,6 +213,55 @@ bool EditBox::handle_mousepress(const uint8_t btn, int32_t, int32_t) {
 	return false;
 }
 
+void EditBox::set_caret_to_cursor_pos(int32_t x) {
+	if (x <= 2 * kMarginX) {
+		set_caret_pos(0);
+		return;
+	}
+
+	int text_w = text_width(m_->text, *m_->font_style, m_->font_scale);
+	if (text_w <= 0) {
+		set_caret_to_cursor_pos(0);
+		return;
+	}
+
+	double x_relative = static_cast<double>(x - 2 * kMarginX - m_->scrolloffset) / text_w;
+	if (x_relative > 1) {
+		set_caret_pos(m_->text.size());
+		return;
+	}
+
+	// initial guess of index which works well already if all characters would be of same width
+	int index = x_relative * m_->text.size();
+
+	// mouse coordinate x=0 means leftmost spot in editbox but text starts with margin -> adjust
+	x -= kMarginX;
+	text_w = calculate_text_width(index);
+
+	index = approximate_cursor(x, text_w, index);
+
+	set_caret_pos(index);
+}
+int EditBox::approximate_cursor(int32_t x, int text_w, int index) const {
```

Comment:
This is no binary search but an incremental approximation, but it should terminate in any case.
Any idea how often the loops has to be run?

Reply:
this depends on string length and char distribution of course. In my tests for input string "iiiiiiiiiimmmmmmmmmmmiiiiiii" ("worst case" regarding char width distribution) it was about 10 loops. With real world strings, the very first approximation is already quite good so the loop only needs to run ~2-3 times :)